Add Opt out to SDK

diegoserranoa opened the pull request:
This PR adds opt out functionality to iOS SDK.

## Changes
- Update smartphone and tablets UI
- Update tests
- Add opt out functionality

## How to test

To enable opt out you should use the newly added method 
```
[Wootric showOptOut:YES];
```
Testing should be straightforward. Just remember the different scenarios:
- Test both smartphones and tablets
- Test selecting all three options:
  - `No, the surveys are fine`
  - `Don't show me a survey for awhile`
  - `Yes, opt me out of all future surveys`

Check the console to see everything is correct
```
a = Account.find_by_account_token 'NPS-1234qwer'
eu = a.end_users.where(email: 'test@email.com').take
ap eu
ap eu.settings
```
Depending on the option you selected, the values should be different.
**No, the surveys are fine:** Nothing should change.
**Don't show me a survey for awhile:** A new decline should be created for the end user.
**Yes, opt me out of all future surveys:** A new decline should be created and the `EndUserSettings` should be something like this
```
#<EndUserSettings:0x000055baaf3eb510> {
                              :id => 71320245,
                     :end_user_id => 71495759,
                       :email_nps => false,
                      :created_at => Wed, 16 May 2018 16:16:36 PDT -07:00,
                      :updated_at => Wed, 16 May 2018 16:16:36 PDT -07:00,
                         :web_nps => false,
                      :mobile_nps => false,
             :force_mobile_survey => nil,
                :force_web_survey => nil,
    :surveys_disabled_by_end_user => true
}
```

- Also check that the opt out button disappears for the social share view (the third view)

## Trello
https://trello.com/c/qc8lu3Xf/16-2-gdpr-ios-sdk

## Note
- I left various commits for easier reviewing. All these will be squashed once this PR has been approved.

diegoserranoa committed:
Opt out: update SDK UI

Changes:
- Update smartphone and tablets UI
- Update tests
- Add opt out functionality

Trello
https://trello.com/c/qc8lu3Xf/16-2-gdpr-ios-sdk

marinhero commented:
Do I need to set a flag on to see the opt-out link?

diegoserranoa commented:
@marinhero nope, we have no flags for mobile haha beware if you're testing on the simulator. It sometimes is slow :/

marinhero commented:
Diego, the feature shall only be available per request. We need the variable to toggle the presence of the opt-out link

diegoserranoa commented:
@marinhero which variable?

marinhero commented:
Something like this https://github.com/Wootric/web-beacon/blob/master/public/js/wootric.js#L13

diegoserranoa commented:
@marinhero thanks! This variable can only be set by the SDKs, right? Or does it come from the account settings?

diegoserranoa commented:
Sorry for not asking all this question earlier. My bad...

diegoserranoa commented:
@marinhero PR updated: I've added a new method `[Wootric showOptOut:YES];` and updated the description also. Thanks for catching this and letting me know!

diegoserranoa commented:
I've also updated the Segment interface so the Segment ios integration can be updated.

marinhero commented:
The feature works on every case. The only thing I recommend is to hide the link once the user interacted with the survey.

![screen shot 2018-05-17 at 2 31 08 pm](https://user-images.githubusercontent.com/316711/40205037-6e2def88-59df-11e8-8b05-6765f6d4d021.png)

^ This shouldn't be allowed

diegoserranoa commented:
@marinhero @DanPitrowiski mentioned that the opt-out option should only be hidden in the last view
https://github.com/Wootric/WootricSDK-Android/pull/58#issuecomment-389604560

marinhero commented:
All right, then that's pretty much it. I'm pretty sure this will be greatly benefited from this https://github.com/Wootric/wootric/pull/2143

marinhero reviewed (approved):
Merge it in!
